components/centraldashboard/app/api.ts

```diff
@@ -40,6 +51,21 @@ export class Api {
     return email;
   }
 
+  /**
+   * Retrieves workgroup info from Profile Controller.
+   */
+  private async getWorkgroup(req: express.Request, user: string): Promise<WorkgroupInfo> {
+    const {profileController} = this;
+    const auth = this.getAuthOption(req);
+    const adminResponse = await profileController.v1RoleClusteradminGet(user, auth);
+    const bindings = await profileController.readBindings(user, undefined, undefined, auth);
+    const namespaces = bindings.body.bindings;
```

Comment:
This seems like it will change the shape of the response given back to the client. It looks like it might be the `referredNamespace` property that needs to be referenced perhaps?

Reply:
Yeah I was planning on the front-end being aware of all these details. I'll make front-end changes tomorrow.

Reply:
Whatever you prefer. If it starts to get pretty big, feel free to break the front-end changes into another PR. I think there are still two considerations though that need to be accounted for.

1. Showing namespaces from other profiles that a user might be a *contributor* to.
2. Showing all namespaces for clusters where identity-awareness is not available. This would be all scenarios other than GKE with IAP at this point so we should probably make sure it maintains consistency with today's behavior.

Reply:
1. Already implemented
2. The current behavior is to attempt to interact with Profile Controller API, if the API fails, then fallback action happens to fetch all namespaces